pages/api/users/index.js

```diff
@@ -6,23 +6,28 @@ import Profile from "../../../models/Profile";
 
 export default async function handler(req, res) {
   await connectMongo();
+
   const directoryPath = path.join(process.cwd(), "data");
-  const files = fs
-    .readdirSync(directoryPath)
-    .filter((item) => item.includes("json"));
+  const files = await fs.promises.readdir(directoryPath);
+
+  const users = await Promise.all(
+    files
+      .filter((item) => item.includes("json"))
+      .map(async (file) => {
+        const filePath = path.join(directoryPath, file);
+        try {
+          const data = await fs.promises.readFile(filePath, "utf8");
+          return {
+            ...JSON.parse(data),
+            username: file.split(".")[0],
+          };
+        } catch (e) {
+          console.log(`ERROR loading profile "${filePath}"`);
+          return null;
```

Comment:
`flatMap` was used to remove any files that did not load, but now there will be a profile that has `null` in it?